packages/app/src/utils/tenderSwapHooks.ts

```diff
@@ -59,6 +60,43 @@ export const useCalculateRemoveLiquidityOneToken = (
   return result?.value?.[0] ?? constants.Zero;
 };
 
+export const usePriceImpact = (
+  isSendingToken: boolean,
+  pool: string,
+  inputAmount: string,
+  receiveAmount: BigNumber
+) => {
+  const swapContract = new Contract(pool, TenderSwapABI) as TenderSwap;
+
+  const resultTenderToken = useCall(
+    pool && {
+      contract: swapContract,
+      method: "getToken0Balance",
+      args: [],
+    }
+  );
+  const resultUnderlyingToken = useCall(
+    pool && {
+      contract: swapContract,
+      method: "getToken1Balance",
+      args: [],
+    }
+  );
+
+  const tokenBalance = resultUnderlyingToken?.value?.[0] ?? constants.One;
+  const tenderTokenBalance = resultTenderToken?.value?.[0] ?? constants.One;
+
+  const spotPrice = isSendingToken
+    ? Number.parseFloat(tokenBalance.mul(1000000).div(tenderTokenBalance).toString()) / 1000000
+    : Number.parseFloat(tenderTokenBalance.mul(1000000).div(tokenBalance).toString()) / 1000000;
+
+  const spotOutputAmount = Number.parseFloat(inputAmount ?? 0) / spotPrice;
+  const priceImpact =
+    (spotOutputAmount - Number.parseFloat(weiToEthWithDecimals(receiveAmount ?? constants.One, 5))) / spotOutputAmount;
```

Comment:
WDYT of creating an actual hook in this file `useExecutionPrice`

and use that here ? 

The price impact would then be 

```
spotPrice <= executionPrice ? 
spotPrice / executionPrice * 100 : - spotPrice / executionPrice * 100
```

Reply:
in that case we could just return the executionPrice from this hook, otherwise we would calculate it twice